zp-relayer/validateTx.ts

```diff
@@ -202,14 +187,30 @@ async function getRecoveredAddress(
   return recoveredAddress
 }
 
-export async function validateTx(
-  { txType, rawMemo, txProof, depositSignature }: TxPayload,
-  delta: Delta,
-  nullifier: string
-) {
+async function checkRoot(index: BN, proofRoot: string, poolSet: RootSet, optimisticSet: RootSet) {
+  const indexStr = index.toString(10)
```

Comment:
deltaIndex is Option<String> actually, so there's a small chance that  it will be `undefined` here. 
may be add a check for that as well?

Reply:
Is it? DeltaIndex is used as public input in SNARK, so it is a required parameter. If user sent bad proof inputs, such error will be thrown here:
https://github.com/zkBob/zeropool-relayer/blob/17af262a3ae146bfbdd4d39fc261bb5e5c89f637/zp-relayer/validateTx.ts#L211
Then, it will be caught by worker, and job will fail (as expected).

Reply:
Yeah, got confused by the function signature
https://github.com/zkbob/libzkbob-rs/blob/6440a3ce8c6050095f8a591ca51cca2bd7f3f7b0/libzeropool-rs/src/client/mod.rs#L128
but it is indeed checked in the proof